lib/inspec/resources/oracle.rb

```diff
@@ -0,0 +1,52 @@
+require "inspec/resources/powershell"
+
+module Inspec::Resources
+  class Oracle < Inspec.resource(1)
+    name "oracle"
+    supports platform: "unix"
+    supports platform: "windows"
+
+    desc "The 'oracle' resource is a helper for the 'oracledb_listener_conf'"
+
+    attr_reader :conf_path
+
+    def initialize
+      case inspec.os[:family]
+      when "debian", "redhat", "linux", "suse"
+        determine_conf_dir_and_path_in_linux
+      when "windows"
+        determine_conf_dir_and_path_in_windows
+      end
+    end
+
+    def to_s
+      "OracleDB"
+    end
+
+    private
+
+    def determine_conf_dir_and_path_in_linux
+      oracle_home = inspec.command("echo $ORACLE_HOME").stdout&.chomp
+      if oracle_home.empty?
+        warn "No oracle listener settings found in $ORACLE_HOME/network/admin directory"
```

Comment:
This did not actually check for the existence of the path.

Reply:
I see your point. I need to have an external check for actually validating the presence of this file. I had a confusion about whether we should be also warning if the ORACLE_HOME value is set or not because if it is not set it will not be able to fetch listener file.

Reply:
Right - you can still use `inspec.command` and `find`, just use ENV['ORACLE_HOME'] as your starting point instead of `/`. 

As for whether ORACLE_HOME is set, ENV is just a hash, you can use `key?`.